src/Microsoft.Extensions.Configuration.AzureAppConfiguration/LocalConfigurationClient.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure.Data.AppConfiguration;
+using System;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal class LocalConfigurationClient
```

Comment:
>LocalConfigurationClient [](http://example.com/codeflow?start=19&length=24)

This new type is introduced for mapping endpoint to client. Can we use a Dictionary instead to create `FailOverSupportedConfigurationClient`?

Reply:
We need two way mapping here. i.e. endpoint <--> client.
The endpoint --> client mapping is required because the `UpdateSyncToken` API provides the endpoint whose sync token needs to be updated, and we need to find the corresponding client and then make the request.

Currently there are not many OOTB data structures that support it. There are a few libraries which support two-way mapping, but they are not ordered.
The best way I could think of is to have this type.

Reply:
Why do we need client -> endpoint mapping?

Reply:
I was thinking we need the client -> endpoint mapping for calling the APIs, but when I looked at the code, that's not the case.
The reason I didn't opt for endpoint -> client dictionary is because in the `FailOverSupportedConfigurationClient` we need ordered list of clients. So we can technically have a `OrderedDictionary` of endpoint -> client.
But C# doesn't have generics support for `OrderedDictionary`. i.e. There is no `OrderedDictionary<U, T>`.
This makes us do type casting everytime we use `ConfigurationClient` and IMO, it is an ugly solution and prone to errors.

I would rather -
1. Have Azure SDK expose `endpoint` in their `ConfigurationClient` type.
2. Or go with the `LocalConfigurationClient` class.

Let me know if you see any pitfalls with this or feel strongly that we should use `OrderedDictionary`.

Reply:
In FailOverSupportedConfigurationClient, you are currently storing a List of LocalConfigurationClients to maintain that insertion order. I was thinking that another option is to use a List of `Tuple<Uri, ConfigurationClient>`. It has the same effect without introducing a new type. But the downside of using a tuple is when you access the clients in the list, you'll have to refer to the endpoint as `Item1` and client as `Item2`. So the code is not that readable. 
I'm fine with the new `LocalConfigurationClient` type. Can you move this class to the Models folder? We have other similar types defined there.

Reply:
That sounds good.
Do you think we should persue Azure SDK to expose `Endpoint` in `ConfigurationClient` type?
Probably not for this iteration, but for the next one?

Reply:
I can't think of any other use for exposing `Endpoint` in `ConfigurationClient` right now. If we have a stronger use case, we can push for it, but doing it just for this implementation doesn't sound compelling.